### ISSUE TITLE
TypeError: "app is undefined" calling function from mounted lifecycle hook

### ISSUE THREAD

**Issue by extra808** [susanBuck/e28-fall19#67]
In my original game, I used `window.addEventListener('load', newGame());` to set the game conditions. Now I'm trying to use the `mounted` lifecycle hook but I'm getting the error `TypeError: "app is undefined"` in the console and the function does not run.

```JavaScript
	methods: {
		checkGuess: function () {
			checkGuess();
		},
		newGame: function () {
			newGame();
		}
	},
	mounted: function () {
		this.newGame();
	},
```

Within the newGame() function below the creation of the Vue object does use app.* to access the keys within `data` and that's worked fine until I try to call the function from the lifecycle hook. `<button id="newGame" v-on:click='newGame()'>New Game</button>` works fine.

If I move the contents of the  newGame() function to within the Vue newGame method and change all the `app.`s too `this.`s, it works but it seems unwieldy to have everything written within the Vue object.

**Comment by susanBuck** [comment]
You could pass `this` to `newGame`:

```js
mounted: function () {
        newGame(this);
},
```

Your `newGame` function outside the scope of the Vue instance would then have access to the Vue instance:
```js
function newGame(appInstance) {
    console.log(appInstance);
}
```

Alternatively, you could define your `newGame` function as a variable outside of the Vue instance, then define a `newGame` method within the Vue Instance that is set to that variable:
```js
let newGameFunc = function newGame() {
    console.log(this);
}

let app = new Vue({
    el: '#app',   
    mounted: function () {
        this.newGame();
    },
    methods: {
        newGame: newGameFunc
    }
});
```

Let me know if the above suggestions touch upon what you're looking for, or if I've misinterpreted the question.

**Comment by extra808** [comment]
Your first suggestion works if I change every instance it's called (`newGame(this);`) and if I change the function to have a parameter (`function newGame(app)`).

All my functions outside the scope of the Vue instance can access the instance by using `app.` just not when the function is called by the `mounted` lifecycle hook; e.g. the functions called by the Vue methods I've shown above access the instance using `app.` without a problem. Maybe the instance doesn't fully exist when `mounted` is triggered?

I've made a number of changes so I've ended up avoiding the issue anyway.
